tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -555,7 +555,7 @@ async def load_image_to_docker_host(context: ConnectorContext, tar_file: File, i
     # Not tagged images only have a sha256 id the load output shares.
     if "sha256:" in image_load_output:
         image_id = image_load_output.replace("\n", "").replace("Loaded image ID: sha256:", "")
-        await docker_cli.with_exec(["docker", "tag", image_id, image_tag]).exit_code()
+        await docker_cli.with_exec(["docker", "tag", image_id, image_tag])
```

Comment:
Is `with_exec` now awaitable? INo need to use `exit_code` or `sync` to get the command executed?

Reply:
Yeah this is not a typo 🙂 It's actually `Container` that's awaitable:
```python
# execute container and return it or raise exception on error
await container

# same as
await container.sync()
```
I like the simple syntax of doing `await` on a lazy object (i.e., `Container`) to execute it. I feel the `.sync()` is redundant here, that's why I tend to drop it. It still returns the same container which allows chaining:
```python
# short circuit on error
out = await (await container).with_exec([...]).stdout()
```

This works for any type that has a `.sync()` method by the way. You can do stuff like this:
```python
# poor man's concurrency 😂
await (
    client.directory()
    .with_directory("/test", test_ctr.rootfs())
    .with_directory("/lint", lint_ctr.rootfs())
)
```

Reply:
Nice! So whenever we want to run `with_exec` commands and not care about the `stdout / stderr` we can directly await the Container